### PR TITLE
refactor(uat): deleted validation and added step set mqtt timeout in MqttControlSteps

### PR DESCRIPTION
**Pull request by auarbekov**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-68
Create validate connection step

**Description of changes:**
Deleted validation step and added step set mqtt timeout in MqttControlSteps

**Why is this change necessary:**
Code of MQTT client based on AWS IoT device SDK should be covered by unit tests.

**How was this change tested:**
At the moment manually, as part of test scenario. In phase 2 we are going to cover by unit tests.

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bgklika** [comment]
1. Please fill PR message using the template automatically provided
2. CI build of project now failed, please take a look to CI logs and resolve the problems

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 1]
```diff
@@ -38,7 +38,7 @@ public class MqttControlSteps {
 
     private static final int DEFAULT_CONTROL_GRPC_PORT = 47_619;
     private static final int DEFAULT_MQTT_KEEP_ALIVE = 60;
-    private static final int DEFAULT_MQTT_CONNECT_TIMEOUT = 30;
+    private static final int DEFAULT_MQTT_TIMEOUT = 30;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:41]: Please do not use that constant directly.
1. Create private variable and initialize it from DEFAULT_MQTT_TIMEOUT.
2. Use variable in any MQTT operation  which requires timeout.
3. Create building block with set that timeout
4. Do not use that building block in steps of 1st scenario.

**Review by bgklika — CHANGES_REQUESTED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 1 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:41]: As has been described in https://klika-tech.atlassian.net/browse/GGMQ-68

1. step "Then connection for device "clientDeviceTest" is successfully established within 3 seconds" must be removed from the scenario ggmq-1.feature
2. building block "@Then("connection for device {string} is successfully established within {int} {word}")" must be removed from MqttControlSteps.java

**Review by esmirnov-klika-tech — DISMISSED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 2]
```diff
@@ -67,6 +67,8 @@ public class MqttControlSteps {
 
     private final GreengrassV2Client greengrassClient;
 
+    private Integer mqttTimeout;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:70]: Should be be simple int,  must be initilized

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 3]
```diff
@@ -166,10 +168,9 @@ public void connect(String clientDeviceId, String componentId, String brokerId)
                      .createMqttConnection(request, connectionEvents);
     }
 
-    @Then("connection for device {string} is successfully established within {int} {word}")
-    public void validateConnect(String clientDeviceId, int value, String unitOfMeasure) {
-        //@TODO Implement method
-        throw new RuntimeException("Method validateConnect is not implemented");
+    @And("I set MQTT timeout {int}")
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:171]: Please add javadoc comment

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 4]
```diff
@@ -258,7 +259,7 @@ private MqttConnectRequest getMqttConnectRequest(String clientDeviceId, String c
                                  .setPort(getBrokerPort(brokerId))
                                  .setKeepalive(DEFAULT_MQTT_KEEP_ALIVE)
                                  .setCleanSession(true)
-                                 .setTimeout(DEFAULT_MQTT_CONNECT_TIMEOUT)
+                                 .setTimeout(getMqttTimeout())
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:262]: Getter for private field is not required

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 5]
```diff
@@ -347,4 +348,9 @@ private String getBrokerCa(String brokerId) {
     private void putBrokerCa(String brokerId, String ca) {
         scenarioContext.put("ca-" + brokerId, ca);
     }
+
+    private int getMqttTimeout() {
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:352]: That method is not required at all

**Review by bgklika — CHANGES_REQUESTED** [review]

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 1 above)
Comment (bgklika) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:56]: Fixed in a0d4b6547583e61e60859894ddacb5cb4b059e0d

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 3 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:176]: This will help set the timeout in the future

**Review by bgklika — APPROVED** [review]

**Pull request closed without merging**